tests/components/ViewedByCustomer/ViewedByCustomer.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { ViewedByCustomer } from '../../../src/Index';
+import { Mock, Fake } from 'coveo-search-ui-tests';
+import { IViewedByCustomerOptions } from '../../../src/components/ViewedByCustomer/ViewedByCustomer';
+import { IQueryResult } from 'coveo-search-ui';
+
+describe('ViewedByCustomer', () => {
+    describe('when the field isUserActionView of the result true', () => {
+        const fakeResult: IQueryResult = { ...Fake.createFakeResult(), isUserActionView: true };
+
+        it('should display the default label and the icon when no options are given', () => {
+            const testComponent = Mock.advancedResultComponentSetup<ViewedByCustomer>(ViewedByCustomer, fakeResult).cmp;
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-icon').length).toBe(1);
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-label').length).toBe(1);
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-label').item(0).innerHTML).toBe('Viewed by Customer');
+        });
+
+        it('should display the label given in the option if any is given', () => {
+            const option = new Mock.AdvancedComponentSetupOptions();
+            (option.cmpOptions as IViewedByCustomerOptions) = { label: 'test label' };
+            const testComponent = Mock.advancedResultComponentSetup<ViewedByCustomer>(ViewedByCustomer, fakeResult, option).cmp;
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-label').length).toBe(1);
+            expect(testComponent.element.getElementsByClassName('viewed-by-customer-label').item(0).innerHTML).toBe('test label');
+        });
+
+        it('should hide the icon if showIcon option is set to false', () => {
+            const option = new Mock.AdvancedComponentSetupOptions();
+            (option.cmpOptions as IViewedByCustomerOptions) = { showIcon: false };
```

Comment:
not sure about the cast, can't you use the constructor instead ?

Reply:
To be fair, the cast is mostly for typing, it does not need to exists.
Why not set directly in the constructor: because I can't only set the option without setting the element because cmpOptions is the second attribute.
https://github.com/coveo/search-ui-tests/blob/master/src/MockEnvironment/AdvancedComponentsSetup.ts#L44

Reply:
yes you can `new Mock.AdvancedComponentSetupOptions(null, { showIcon: false });`